--- src/mapi/glapi/meson.build.orig
+++ src/mapi/glapi/meson.build
@@ -76,6 +76,7 @@
   c_args : [c_msvc_compat_args, static_glapi_args],
   dependencies : [dep_thread, dep_selinux],
   build_by_default : false,
+  install : true,
 )
 
 if with_any_opengl and not with_shared_glapi and with_tests
--- src/mapi/shared-glapi/meson.build.orig
+++ src/mapi/shared-glapi/meson.build
@@ -36,7 +36,7 @@
   capture : true,
 )
 
-libglapi = shared_library(
+libglapi = both_libraries(
   'glapi',
   [files_mapi_glapi, files_mapi_util, shared_glapi_mapi_tmp_h],
   c_args : [
